tests/test_inference.py

```diff
@@ -652,6 +653,34 @@ def process_line(word_pos):
             )
         )
 
+    def test_fstring_inference(self) -> None:
+        code = """
+            name = "John"
+            result = f"Hello {name}!"
+            """
+        ast = parse(code, __name__)
+        node = ast["result"]
+        inferred = node.inferred()
+        self.assertEqual(len(inferred), 1)
+        value_node = inferred[0]
+        self.assertIsInstance(value_node, Const)
+        self.assertEqual(value_node.value, "Hello John!")
+
+    def test_formatted_fstring_inference(self) -> None:
+        code = """
+            width = 10
+            precision = 4
+            value = 12.34567
+            result = f"result: {value:{width}.{precision}}!"
+            """
+        ast = parse(code, __name__)
+        node = ast["result"]
+        inferred = node.inferred()
+        self.assertEqual(len(inferred), 1)
+        value_node = inferred[0]
+        self.assertIsInstance(value_node, Const)
+        self.assertEqual(value_node.value, "result:      12.35!")
+
```

Comment:
Please also update the assertion in test_no_crash_on_const_reconstruction.

Reply:
Actually, that will be moot after https://github.com/pylint-dev/astroid/pull/2484.

Reply:
Done anyway